### PR TITLE
Modified CI workflows

### PR DESCRIPTION
**Pull request by nidhi-nair**
## ~_Anyone glancing through this PR at any point, please leave an approval to trigger a test of this workflow. Also, please don't merge this PR in even if it is already in an Approved state._~ 😄 This is handled by modifying a Cypress test

Changes to workflow now follow the following rules for execution:
- When there is a change in a PR for client side code, the client code will build and run jest tests using the PR `HEAD`
- When there is a change in a PR for server side code, the server code will build and run unit tests using the PR `HEAD`
- When there is an approval for a PR, the server code will build, run unit tests and create a docker image in that instance. The client side code will also build and run unit tests. The integration tests will use the local docker image to run cypress tests. All these builds will use the internal PR merge SHA
- When there is a merge into `release` or `master`, the integration workflow will run, this time using the branch `HEAD`, which might be different from the merge SHA in the previous condition

Notes:
Should I use the merge SHA for unit tests as well?

### CODE REVIEW

**Review by hetunandu — APPROVED** [review]

**Comment by hetunandu** [comment]
I was thinking of one more case for running the integration test:
if any files changes in the integration test folder `app/client/cypress`, it should run the integration workflow instead of the individual build workflow. How complicated would that make this change?

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -1,25 +1,134 @@
-name: Appsmith Client Test Workflow
+name: Appsmith Integration Workflow
 
 on:
   # This line enables manual triggering of this workflow.
   workflow_dispatch:
 
+  push:
+    branches: [release, master]
+
```
Comment (hetunandu) [inline .github/workflows/integration.yml:9]:
```suggestion
  push:
    branches: [release, master]
 
  pull_request:
    branches: [release]
    paths:
      - 'app/client/cypress/**'
      - '!app/client/cypress/manual_TestSuite/**'
```

**Comment by nidhi-nair** [comment]
That's doable, and sensible. I think that will nullify all of these changes though :D Don't all client side changes include Cypress tests? @hetunandu

**Comment by hetunandu** [comment]
No, not all. Many won't. That is part of the problem 🙃

File: .github/workflows/integration.yml (hunk 1 above)
Comment (nidhi-nair) [reply .github/workflows/integration.yml:9]: This will need to be paired with proper docker images also, I'll push this from local

Comment (nidhi-nair) [reply .github/workflows/integration.yml:9]: Curious, what is inside of `manual_TestSuite`? @hetunandu

Comment (hetunandu) [reply .github/workflows/integration.yml]: Those are test ideas written by @prapullac which she uses in her workflow with the goal of eventually implementing them via automation.

**Review by hetunandu — DISMISSED** [review]

**Comment by mohanarpit** [comment]
@nidhi-nair Is this PR still required? If not, can we close this?

**Pull request closed without merging**